docker/mysql/docker-compose.yml

```diff
@@ -0,0 +1,106 @@
+version: '2.3'
+
+services:
+
+  conductor-server:
+    environment:
+      - CONFIG_PROP=config-mysql.properties
+    image: conductor:server
+    build:
+      context: ../../
+      dockerfile: docker/server/Dockerfile
+    networks:
+      - internal
+    links:
+      - elasticsearch:es
+    ports:
+      - 8080:8080
+      - 8090:8090
+    depends_on:
+      elasticsearch:
+        condition: service_healthy
+      mysql:
+        condition: service_healthy
+    volumes:
+      - ../server/config:/app/config
+      - ./logs:/app/logs
+
+  conductor-ui:
+    environment:
+      - WF_SERVER=http://conductor-server:8080/api/
+    image: conductor:ui
+    build:
+      context: ../../
+      dockerfile: docker/ui/Dockerfile
+    networks:
+      - internal
+    ports:
+      - 5000:5000
+    depends_on:
+      - conductor-server
+
+  mysql:
+    image: mysql:5.7 # 5.6 and below is not supported
+    environment:
+      MYSQL_ROOT_PASSWORD: 12345
+      MYSQL_DATABASE: conductor
+      MYSQL_USER: conductor
+      MYSQL_PASSWORD: conductor
+    volumes:
+      - type: volume
+        source: conductor_mysql
+        target: /var/lib/mysql
+    networks:
+      - internal
+    ports:
+      - 3306:3306
+    healthcheck:
+      test: timeout 5 bash -c 'cat < /dev/null > /dev/tcp/localhost/3306'
+      interval: 5s
+      timeout: 5s
+      retries: 12
+
+  # https://www.elastic.co/guide/en/elasticsearch/reference/5.6/docker.html
+  # if Elasticsearch Container Stopped with Exit 78
+  # run `sudo sysctl -w vm.max_map_count = 262144` in terminal 
+  # https://github.com/laradock/laradock/issues/1699
+  elasticsearch:
+    image: elasticsearch:6.8.15
+    environment:
+      - cluster.name=docker-cluster
+      - bootstrap.memory_lock=true
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+    networks:
+      - internal
+    ports:
+      - 9200:9200
+      - 9300:9300
+    healthcheck:
+      test: timeout 5 bash -c 'cat < /dev/null > /dev/tcp/localhost/9300'
+      interval: 5s
+      timeout: 5s
+      retries: 12
+
+  elasticsearch2:
```

Comment:
Could you please document the need for this section.

Reply:
For sure.
Those configuration are copied from elastic multi-node configuration,
https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html#docker-compose-file

I will tag the link in.

Reply:
Added documentation link

Reply:
Since these docker compose files are used for local testing, a multi-node elasticsearch cluster setup would not be needed, rather a single node elasticsearch cluster should suffice.

Reply:
Agree. But without multi-node setup. The next startup after the first will not be run successfully because it will timeout, multi-node to solve the little gotcha. Related to #2262

Reply:
I understand the intention here and the steps taken, but it adds an extra layer of complexity for a problem that requires a small tweak in the properties  file (config-mysql.properties) `conductor.elasticsearch.clusteHealthColor=yellow`. 
Keeping the docker files lean with lesser dependencies will help users to get up and running quicker.

Additionally, since local elasticsearch cluster does not reflect a production use-case and is only used for debugging/onboarding, an elaborate setup would be overkill.

Reply:
Consider making the property change above and removing this multi-node setup.

Reply:
Didn't aware of `conductor.elasticsearch.clusteHealthColor=yellow`, definitely will test again. It could be a big help. Thank in advance!

Reply:
I have tried with above config, it seems that it will be able to run for the first time, but second time it will generate 
```
71458 [main] WARN  org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext [] - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'workflowBulkResource' defined in URL [jar:file:/app/libs/conductor-server-3.1.0-SNAPSHOT-boot.jar!/BOOT-INF/lib/conductor-rest-3.1.0-SNAPSHOT.jar!/com/netflix/conductor/rest/controllers/WorkflowBulkResource.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'workflowBulkServiceImpl' defined in URL [jar:file:/app/libs/conductor-server-3.1.0-SNAPSHOT-boot.jar!/BOOT-INF/lib/conductor-core-3.1.0-SNAPSHOT.jar!/com/netflix/conductor/service/WorkflowBulkServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'workflowExecutor' defined in URL [jar:file:/app/libs/conductor-server-3.1.0-SNAPSHOT-boot.jar!/BOOT-INF/lib/conductor-core-3.1.0-SNAPSHOT.jar!/com/netflix/conductor/core/execution/WorkflowExecutor.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'executionDAOFacade' defined in URL [jar:file:/app/libs/conductor-server-3.1.0-SNAPSHOT-boot.jar!/BOOT-INF/lib/conductor-core-3.1.0-SNAPSHOT.jar!/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.class]: Unsatisfied dependency expressed through constructor parameter 2; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'es6IndexDAO': Invocation of init method failed; nested exception is java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]
71481 [main] INFO  com.zaxxer.hikari.HikariDataSource [] - HikariPool-1 - Shutdown initiated...
71492 [main] INFO  com.zaxxer.hikari.HikariDataSource [] - HikariPool-1 - Shutdown completed.
71499 [main] INFO  org.apache.catalina.core.StandardService [] - Stopping service [Tomcat]
71531 [main] INFO  org.springframework.boot.autoconfigure.logging.ConditionEvaluationReportLoggingListener [] - 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
71533 [main] ERROR org.springframework.boot.SpringApplication [] - Application run failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'workflowBulkResource' defined in URL [jar:file:/app/libs/conductor-server-3.1.0-SNAPSHOT-boot.jar!/BOOT-INF/lib/conductor-rest-3.1.0-SNAPSHOT.jar!/com/netflix/conductor/rest/controllers/WorkflowBulkResource.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'workflowBulkServiceImpl' defined in URL [jar:file:/app/libs/conductor-server-3.1.0-SNAPSHOT-boot.jar!/BOOT-INF/lib/conductor-core-3.1.0-SNAPSHOT.jar!/com/netflix/conductor/service/WorkflowBulkServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'workflowExecutor' defined in URL [jar:file:/app/libs/conductor-server-3.1.0-SNAPSHOT-boot.jar!/BOOT-INF/lib/conductor-core-3.1.0-SNAPSHOT.jar!/com/netflix/conductor/core/execution/WorkflowExecutor.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'executionDAOFacade' defined in URL [jar:file:/app/libs/conductor-server-3.1.0-SNAPSHOT-boot.jar!/BOOT-INF/lib/conductor-core-3.1.0-SNAPSHOT.jar!/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.class]: Unsatisfied dependency expressed through constructor parameter 2; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'es6IndexDAO': Invocation of init method failed; nested exception is java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]
```

Reply:
My elastic docker config
```
 elasticsearch:
    image: elasticsearch:6.8.15
    environment:
      - "ES_JAVA_OPTS=-Xms512m -Xmx1024m"
      - transport.host=0.0.0.0
      - discovery.type=single-node
      - xpack.security.enabled=false
    networks:
      - internal
    ports:
      - 9200:9200
      - 9300:9300
    healthcheck:
      test: timeout 5 bash -c 'cat < /dev/null > /dev/tcp/localhost/9300'
      interval: 5s
      timeout: 5s
      retries: 12
    logging:
      driver: "json-file"
      options:
        max-size: "1k"
        max-file: "3"
```

Reply:
@apanicker-nflx I have removed the cluster. And also found the config that causing it to fail. 

It seems that `conductor.metrics-logger.enabled=true` causing 
`Error creating bean with name 'workflowBulkResource'`. I have removed it, and it runs fine, which i dont understand, because first run index is created, but after second run, The error occurs, but this docker-compose should run fine.

Reply:
If you want to read the error message it is at the commit message: https://github.com/imcyee/conductor/commit/4821c8d0dedbc618508b807435db7aa8d3bfa8fd